Content.Client/Construction/UI/FlatpackCreatorMenu.xaml.cs

```diff
@@ -70,11 +70,17 @@ protected override void FrameUpdate(FrameEventArgs args)
         else if (_currentBoard != null)
         {
             //todo double trycomp is kinda stinky.
-            if (_entityManager.TryGetComponent<MachineBoardComponent>(_currentBoard, out var board) &&
-                board.Prototype != null)
+            if (_entityManager.TryGetComponent<MachineBoardComponent>(_currentBoard, out var machineBoardComp) &&
+                machineBoardComp.Prototype != null)
             {
                 var cost = _flatpack.GetFlatpackCreationCost((_owner, flatpacker),
-                    (_currentBoard.Value, board));
+                    (_currentBoard.Value, machineBoardComp));
+                PackButton.Disabled = !_materialStorage.CanChangeMaterialAmount(_owner, cost);
+            }
+            else if (_entityManager.TryGetComponent<ComputerBoardComponent>(_currentBoard, out var computerBoardComp) &&
+                     computerBoardComp.Prototype != null)
+            {
+                var cost = _flatpack.GetFlatpackCreationCostForComputer((_owner, flatpacker));
```

Comment:
you can reduce a lot of duplication like this by just having an if statement that gets the cost and then assigning it to a variable defined beforehand

PSEUDOCODE WARNING:
```cs
Dictionary<string, int>? cost = null;

if (machineBoard)
{
    cost = GetCost(machineBoard);
}
else if (computerBoard)
{
    cost = GetCost(computerBoard);
}

if (cost == null)
   return;

// do duplicated logic here.
```